launch/launch/actions/execute_local.py

```diff
@@ -597,6 +599,10 @@ def prepare(self, context: LaunchContext):
             # pid is added to the dictionary in the connection_made() method of the protocol.
         }
 
+        if self.__respawn:
+            self.__respawn = perform_typed_substitution(
+                context, normalize_typed_substitution(self.__respawn, bool), bool)
```

Comment:
@kosuke55 consider normalizing the substitution in the constructor (so that it fails early if malformed).

Reply:
@hidmic Chaged to normalizing it in the constructor. https://github.com/ros2/launch/pull/569/commits/edb8924c48343b7ca25225a414a40a4275d50af7